browser/scripts/renderer.js

```diff
@@ -391,8 +364,35 @@ Renderer.prototype.set_fullscreen = function(state)
 
 Renderer.prototype.update_viewport = function()
 {
+
 	var c = this.canvas[0];
-	this.context.viewport(0, 0, c.width, c.height);
+	var that = this;
+
+	var canvasArea = E2.app.calculateCanvasArea();
```

Comment:
E2.app.calculateCanvasArea might not be visible eg. in the case of a standalone player. Maybe put a stub of it in player.js CreatePlayer where it creates an empty E2.app?